### PR TITLE
Feat : Added auth

### PR DESCRIPTION
**Pull request by srejitk**
This Merge Request adds the following:

1. Renamed FilterProvider to ProductProvider
2. Added Auth
3. Added the initial steps for Wishlist and Cart
4. Added Login and SIgnUp Pages

### CODE REVIEW

**Review by coldpigli — APPROVED** [review]
Really liked the CSS naming conventions. Would only suggest to remove comments.
Other than that everything looks good

File: House-of-glass/src/components/Product List/ProductList.jsx [hunk 1]
```diff
@@ -1,26 +1,25 @@
...
+import Filters from "components/Filters/Filters";
+import { useProduct } from "components/Filters/ProductContext";
+import ProductCard from "components/Cards/Vertical/ProductCard";
 
 export default function ProductList() {
-
-    const {finalProductList} = useFilter();
-
-
+  const { finalProductList } = useProduct();
```
Comment (SimranjitKaurMaan) [inline House-of-glass/src/components/Product List/ProductList.jsx]: Just `products` or `productList` is enough

**Review by SimranjitKaurMaan — APPROVED** [review]
Better variable naming can be done. Rest looks good.

File: House-of-glass/src/components/Product List/ProductList.jsx (hunk 1 above)
Comment (srejitk) [reply House-of-glass/src/components/Product List/ProductList.jsx]: Thank You Simran. Updated in the latest commit.

**Pull request merged**